server/controllers/profile.js

```diff
@@ -0,0 +1,93 @@
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile
+// @desc Create user profile with valid token
+// @access Private
+exports.createProfile = asyncHandler(async (req, res, next) => {
+  const body = req.body;
+  const id = body.userId;
+  if (!id) {
+    res.status(401);
+    throw new Error("Not Authorized");
+  }
+
+  const existingProfile = await Profile.findOne({ userId: [id] });
+  if (!existingProfile) {
+    res.status(400);
+    throw new Error("Profile already exists");
+  }
+
+  const user = await User.findById(body.userId);
+
+  //create new profile
+  if (user) {
+    const profile = await Profile.create({
+      userId: user._id,
+      firstName: body.firstName,
+      lastName: body.lastName,
+      description: body.description,
+      profilePic: body.profilePic,
+      dateOfBirth: body.dateOfBirth,
+      phone: body.phone,
+      address: body.address,
+      location: body.location,
+      gender: body.gender,
+      isSitter: body.isSitter,
+      hourlyRate: body.hourlyRate,
+      availability: body.availability,
+    });
+    if (!newProfile) {
+      res.status(400);
```

Comment:
```suggestion
      res.status(500);
```
This will be a server error

Reply:
Change made.